core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperatorTableFactory.java

```diff
@@ -70,18 +71,44 @@ private SqlLibraryOperatorTableFactory(Class... classes) {
   /** A cache that returns an operator table for a given library (or set of
    * libraries). */
   @SuppressWarnings("methodref.receiver.bound.invalid")
-  private final LoadingCache<ImmutableSet<SqlLibrary>, SqlOperatorTable> cache =
+  private final LoadingCache<CacheKey, SqlOperatorTable> cache =
       CacheBuilder.newBuilder().build(CacheLoader.from(this::create));
 
+  /**
+   * Cache key that contains library set and a flag indicate whether include operators under
```

Comment:
Can you explain why cacheKey is required? After expanding 'all', the set of SqlLibary is canonical.

Reply:
Because I want to handle the special case in `DocumentationTest#testAllFunctionsAreDocumented`, e.g. we do not want to include the operators under 'ALL' when we get operators under 'SPARK'.